research/format.markdown

```diff
@@ -0,0 +1,99 @@
+# Research on `format` keyword 
+Changes in `format` Keyword Behaviour Across all JSON Schema Drafts: Additions, Deletions, and Upgrades.
+
+## Draft 0 to Draft 2 
+No changes
+
+## Draft 2 to Draft 3
+
+References: [Draft 2 Section 5.20](https://json-schema.org/draft-02/draft-zyp-json-schema-02.txt), [Draft 3 Section 5.23](https://json-schema.org/draft-03/draft-zyp-json-schema-03.pdf)
+
+> Behaviour:   
+      >>[Before](https://json-schema.org/draft-02/draft-zyp-json-schema-02.txt):      
+        Not required to validate - Which means the `format` keyword isn't supposed to do validation by default.  
+        [After](https://json-schema.org/draft-03/draft-zyp-json-schema-03.pdf):   
+        MAY validate (but aren’t required to) - Default v   alidation or no validation are implementation dependent.
+
+| Change Type | Format Attribute | Description                                                      |
+|-------------|------------------|------------------------------------------------------------------|
+| Deleted     | `street-address`  | This SHOULD be a street address                               |
+| Deleted     | `locality`         | This SHOULD be a city or town                                 |
+| Deleted     | `region`           | This SHOULD be a region (a state in the US, province in Canada, etc.) |
+| Deleted     | `postal-code`      | This SHOULD be a postal code (AKA zip code)                    |
+| Deleted     | `country`          | This SHOULD be the name of a country|
+| Added       | `host-name`        | This SHOULD be a host-name                                     |
```

Comment:
@Era-cell I think you are still missing some changes. For example, `host-name` was eventually replaced by `hostname` (which is in i.e. 2020-12) but this document doesn't mention when this happened. Are you sure you didn't miss any other entry like this one?

Reply:
Okay, I didn't consider that as a change. This syntax of the keyword was changed in draft4. I have re-checked everything again and corrected.